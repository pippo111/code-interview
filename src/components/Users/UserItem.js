import React from 'react'
import PropTypes from 'prop-types'

const UserItem = ({ id, name, username }) => (
  <>
    <span className='main-content'>{name}</span>
    <span className='sub-content'>@{username}</span>
  </>
)

UserItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired
}

export default UserItem
