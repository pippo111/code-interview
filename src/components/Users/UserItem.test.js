/* global describe, test, expect */
import React from 'react'
import renderer from 'react-test-renderer'

import UserItem from './UserItem'

describe('UserItem', () => {
  test('snapshot renders', () => {
    const user = { id: 1, name: 'Test', username: 'test' }
    const component = renderer.create(<UserItem {...user} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
