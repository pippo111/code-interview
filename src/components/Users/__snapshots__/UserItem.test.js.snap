// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem snapshot renders 1`] = `
Array [
  <span
    className="main-content"
  >
    Test
  </span>,
  <span
    className="sub-content"
  >
    @
    test
  </span>,
]
`;
